oidc/troubleshooting.md

```diff
@@ -0,0 +1,198 @@
+# SSO Troubleshooting for Howso Platform
+
+This guide provides steps for troubleshooting Single Sign-On (SSO) issues with the Howso Platform. 
+
+## Overview 
+
+When set up to SSO using an external Identity Provider (IdP), Howso Platform uses the Authorization Code Flow with OpenID Connect (OIDC) for user authentication and authorization, optionally implementing Proof Key for Code Exchange (PKCE).
+
+The Howso Platform configuration is under the `oidc` section in the Helm values.  Additional configuration is required within the IdP, to create an OAuth application that Howso Platform can use to authenticate users.
+
+> Note: The Howso Platform User Management Service is itself an IdP and Authorization server, where other internal Howso Platform services are configured as applications.  This is a separate topic.
+
+Once the application is created within the IdP, and the Helm values are applied, unauthenticated users navigating to Howso Platform domains should see the following sign-on button.
+
+<img src="../assets/oidc-sso-login.png" alt="OIDC Authentication Error" width="300" />
+
+Clicking will kick off the authentication flow.  Understanding the main steps of the flow can help with troubleshooting:
+- User navigates to Howso Platform and clicks the continue button to sign-on.
+- Howso Platform redirects the user's browser to the `authorizeEndpoint` specified in the OIDC configuration.  This should bring up the IdP login page. 
+- Once the user has authenticated, the IdP redirects the user back to the `callback` URL in the Howso Platform with an authorization code.  This callback URL is sent to the IdP in the original redirect, but is also part of the configuration supplied to the IdP when the OAuth application is created.  The IdP will check that the redirect URL in the query string and the application configuration match. 
+- If the authentication was successful, the redirect back to Howso Platform will include the authorization code.  Alternatively error information will be included in the query string.
+- On success, the User Management Service component of Howso Platform, the will then exchange this code for an access token and id token using the `tokenEndpoint` and use it to authenticate the user.
+- The User Management Service will then use the `userinfoEndpoint` to get information about the user, such as email address and groups.
+- The access token and id token are validated against the public key provided by the IdP at the `jwksEndpoint`.
+
+
+## Configuration Verification
+
+Ensure all OIDC configuration items are correctly set in your Howso Platform Helm values:
+
+```yaml
+oidc:
+  enabled: true
+  clientID: "<your-client-id>"
+  clientSecret: "<your-client-secret>"
+  algorithm: "RS256"
+  jwksEndpoint: "<jwks-endpoint-url>"
+  authorizeEndpoint: "<authorize-endpoint-url>"
+  tokenEndpoint: "<token-endpoint-url>"
+  userinfoEndpoint: "<userinfo-endpoint-url>"
+  scopes: "<requested-scopes>"
```

Comment:
Should we update this line to scopes: "openid email profile" ?  since that's the exact list of scopes we want our users to use.